### PR TITLE
Dashboard links & service filter

### PR DESCRIPTION
**Pull request by thomasmaas**
Preparation for vertical nav switch:

- [x] Add some missing links to dashboard
- [x] Start using icons for important concepts
- [x] Check for permissions
- [x] Introduce API filter box (@josemigallas)

Next iteration:
- [ ] Up the contrast of the links (maybe later when we actually switch to vertical nav)
- [ ] Refactor dashboard collection link in helper
- [ ] See if we can reduce for loops in filter js
 
screenshot 2018-10-05 22 21 10

filtering: 

screenshot 2018-10-05 22 21 15

### CODE REVIEW

File: app/helpers/provider/admin/dashboards_helper.rb [hunk 1]
```diff
@@ -19,8 +19,13 @@ def sign_class(widget)
     widget.percentual_change > 0 ? 'u-plus' : 'u-minus'
   end
 
-  def dashboard_collection_link(singular_name, collection, path, plural = nil)
-    link_to pluralize(number_to_human(collection.size), singular_name, plural), path, class: 'DashboardNavigation-link'
+  def dashboard_collection_link(singular_name, collection, path, plural = nil, icon_name = nil)
+    link_to path, class: 'DashboardNavigation-link' do
+      link = ''
+      link << icon(icon_name) + ' ' if icon_name
```
Comment (hallelujah) [inline app/helpers/provider/admin/dashboards_helper.rb]: This will be forbidden in future ruby or with `frozen_string_literal: true`

Comment (thomasmaas) [reply app/helpers/provider/admin/dashboards_helper.rb]: yes, rightly so, it's ugly as hell

Comment (thomasmaas) [reply app/helpers/provider/admin/dashboards_helper.rb]: I'm also considering refactoring this method to use named parameters. would be next iteration

File: app/helpers/provider/admin/dashboards_helper.rb [hunk 2]
```diff
@@ -19,8 +19,13 @@ def sign_class(widget)
...
+      link << pluralize(number_to_human(collection.size), singular_name, plural)
+      link.html_safe
```
Comment (hallelujah) [inline app/helpers/provider/admin/dashboards_helper.rb]:
```ruby
link = pluralize(number_to_human(collection.size), singular_name, plural)
link.prepend "#{icon(icon_name)} " if icon_name
```

File: app/views/provider/admin/dashboards/show.html.slim [hunk 3]
```diff
@@ -3,14 +3,15 @@
...
+  = javascript_pack_tag 'dashboard'
 
 .Dashboard
   div.Dashboard-widgets
     // account level
     - if can?(:manage, :partners)
       section.DashboardSection.DashboardSection--audience class=('DashboardSection--wide' unless can?(:manage, :plans))
         header.DashboardSection-header
-          h1.DashboardSection-title Developers
+          h1.DashboardSection-title Audience
```
Comment (guicassolato) [inline app/views/provider/admin/dashboards/show.html.slim]: Still not convinced by “Audience”

Comment (thomasmaas) [reply app/views/provider/admin/dashboards/show.html.slim]: Passed that station

File: app/helpers/provider/admin/dashboards_helper.rb [hunk 4]
```diff
@@ -19,8 +19,13 @@ def sign_class(widget)
     widget.percentual_change > 0 ? 'u-plus' : 'u-minus'
   end
 
-  def dashboard_collection_link(singular_name, collection, path, plural = nil)
-    link_to pluralize(number_to_human(collection.size), singular_name, plural), path, class: 'DashboardNavigation-link'
+  def dashboard_collection_link(singular_name, collection, path, plural = nil, icon_name = nil)
```
Comment (hallelujah) [inline app/helpers/provider/admin/dashboards_helper.rb]: probably use options for more than 3 parameters

File: app/views/provider/admin/dashboards/_services_filter.html.slim [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+javascript:
+  document.addEventListener('DOMContentLoaded', function () {
+    var allApis = #{json @services}
+
+    var displayApis = function(filteredApis) {
+      for (api = 0; api < allApis.length; api++) {
+        document.getElementById("service_" + allApis[api].service.id).style.display = "none"
+      }
+      for (api = 0; api < filteredApis.length; api++) {
```
Comment (hallelujah) [inline app/views/provider/admin/dashboards/_services_filter.html.slim]: Can we avoid doing second loop?
including an if in line 9?

File: app/views/provider/admin/dashboards/show.html.slim [hunk 6]
```diff
@@ -20,14 +21,22 @@
         // Account Level Widgets
         = dashboard_widget :new_accounts
         = dashboard_widget :potential_upgrades if can?(:manage, :plans)
+    section.DashboardSection.DashboardSection--audience class=('DashboardSection--wide' unless can?(:manage, :plans))
+      - if current_user.accessible_services.size > 1
```
Comment (hallelujah) [inline app/views/provider/admin/dashboards/show.html.slim]: Hmm do not use `size` as it makes a count in DB
Prefer using `current_user.accessible_services?`

```ruby

def accessible_services?
  accessible_services.exists?
end
```

File: app/helpers/provider/admin/dashboards_helper.rb (hunk 4 above)
Comment (hallelujah) [reply app/helpers/provider/admin/dashboards_helper.rb]: Another PR for the `dashboard_collection_link` refactoring is accepted 👍

File: app/views/provider/admin/dashboards/_services_filter.html.slim (hunk 5 above)
Comment (thomasmaas) [reply app/views/provider/admin/dashboards/_services_filter.html.slim]: Lets leave this for next iteration @josemigallas

**Comment by thomasmaas** [comment]
In spite of all the red above, it is green. https://jenkins-aws.3sca.net/job/system-upstream/job/pipeline-mysql/557/

**Comment by didierofrivia** [comment]
@thomasmaas it's there, you need to scroll down 👍

File: app/views/provider/admin/dashboards/_services_filter.html.slim (hunk 5 above)
Comment (josemigallas) [reply app/views/provider/admin/dashboards/_services_filter.html.slim]: Problem with that is performance @hallelujah -> http://jsben.ch/XPYGU it's simply much quicker to iterate once each array than looking inside `filteredApis` each time.

**Pull request merged**